widget/embedded/src/store/notification.ts

```diff
@@ -1,26 +1,33 @@
 import type { Notification } from '../types/notification';
-import type {
-  Route,
-  RouteEvent,
-  StepEvent,
-} from '@rango-dev/queue-manager-rango-preset';
 
+import {
+  type PendingSwapWithQueueID,
+  type Route,
+  type RouteEvent,
+  type StepEvent,
+} from '@rango-dev/queue-manager-rango-preset';
 import { create } from 'zustand';
 import { persist, subscribeWithSelector } from 'zustand/middleware';
 
+import { isSwapFinished } from '../utils/swap';
+
 import createSelectors from './selectors';
 
 export interface NotificationState {
+  isSynced: boolean;
   notifications: Notification[];
   setNotification: (event: RouteEvent | StepEvent, route: Route) => void;
   setAsRead: (requestId: Notification['requestId']) => void;
```

Comment:
I think it is better to change `setAsRead` to `removeNotification`.
and remove the read property from the Notification type.